2025.4.1: four tests fails

Four tests fails when I run tests for calver 2025.4.1:
```
$ wget https://files.pythonhosted.org/packages/source/c/calver/calver-2025.4.1.tar.gz
$ gtar xf calver-2025.4.1.tar.gz
$ cd calver-2025.4.1
$ PYTEST_DISABLE_PLUGIN_AUTOLOAD=1 SOURCE_DATE_EPOCH=1743498000 pytest
============================= test session starts ==============================
platform sunos5 -- Python 3.9.21, pytest-8.3.3, pluggy-1.5.0
rootdir: /tmp/test/calver-2025.4.1
configfile: pyproject.toml
collected 8 items                                                              

tests/test_integration.py ...FFF.F                                       [100%]

=================================== FAILURES ===================================
______________________________ test_version_true _______________________________

dist = <stub(metadata=<stub(version='2025.4.1')>)>, keyword = <stub()>

    def test_version_true(dist, keyword):
        value = True

        calver.integration.version(dist, keyword, value)

>       assert dist.metadata.version == datetime.datetime.now().strftime(
            calver.integration.DEFAULT_FORMAT
        )
E       AssertionError: assert '2025.4.1' == '2025.04.02'
E         
E         - 2025.04.02
E         ?      -  ^^
E         + 2025.4.1
E         ?        ^

tests/test_integration.py:44: AssertionError
_______________________________ test_version_str _______________________________

dist = <stub(metadata=<stub(version='2025.4.1')>)>, keyword = <stub()>

    def test_version_str(dist, keyword):
        value = "%c"

        calver.integration.version(dist, keyword, value)

>       assert dist.metadata.version == datetime.datetime.utcnow().strftime(value)
E       AssertionError: assert '2025.4.1' == 'Wed Apr  2 06:19:19 2025'
E         
E         - Wed Apr  2 06:19:19 2025
E         + 2025.4.1

tests/test_integration.py:54: AssertionError
____________________________ test_version_callable _____________________________

dist = <stub(metadata=<stub(version='2025.4.1')>)>, keyword = <stub()>

    def test_version_callable(dist, keyword):
        v = pretend.stub()
        value = lambda: v

        calver.integration.version(dist, keyword, value)

>       assert dist.metadata.version == v
E       AssertionError: assert '2025.4.1' == <stub()>
E        +  where '2025.4.1' = <stub(version='2025.4.1')>.version
E        +    where <stub(version='2025.4.1')> = <stub(metadata=<stub(version='2025.4.1')>)>.metadata

tests/test_integration.py:63: AssertionError
___________________________ test_reproducible_build ____________________________

dist = <stub(metadata=<stub(version='2025.4.1')>)>, keyword = <stub()>
source_date_epoch = 1234567890

    def test_reproducible_build(dist, keyword, source_date_epoch):
        calver.integration.version(dist, keyword, True)

>       assert dist.metadata.version == datetime.datetime.fromtimestamp(
            source_date_epoch, tz=datetime.timezone.utc
        ).strftime(calver.integration.DEFAULT_FORMAT)
E       AssertionError: assert '2025.4.1' == '2009.02.13'
E         
E         - 2009.02.13
E         + 2025.4.1

tests/test_integration.py:80: AssertionError
=========================== short test summary info ============================
FAILED tests/test_integration.py::test_version_true - AssertionError: assert '2025.4.1' == '2025.04.02'
FAILED tests/test_integration.py::test_version_str - AssertionError: assert '2025.4.1' == 'Wed Apr  2 06:19:19 2025'
FAILED tests/test_integration.py::test_version_callable - AssertionError: assert '2025.4.1' == <stub()>
FAILED tests/test_integration.py::test_reproducible_build - AssertionError: assert '2025.4.1' == '2009.02.13'
========================= 4 failed, 4 passed in 0.18s ==========================
$
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -5,7 +5,7 @@ env:
   FORCE_COLOR: 1
 
 jobs:
-  test:
+  build:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -1,0 +1,29 @@
+name: Build
+on: [push, pull_request, workflow_dispatch]
+
+env:
+  FORCE_COLOR: 1
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install build
+    - name: Build package
+      run: python -m build
+    - name: Unpack sdist
+      run: |
+        mkdir -p dist/calver
+        tar -xvvf dist/*.tar.gz -C dist/calver/ --strip-components=1
+    - name: Install tox
+      run: python -m pip install tox
+    - name: Test
+      run: python -m tox -e py --root dist/calver

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,3 +1,4 @@
 include pyproject.toml
+include tox.ini
 include *.md
 include LICENSE

--- a/tests/test_integration.py
+++ b/tests/test_integration.py
@@ -22,6 +22,16 @@ def keyword():
 
 
 @pytest.fixture
+def ignore_pkginfo(monkeypatch):
+    # Ensure that the test doesn't unintently prefer to read from a PKG_INFO
+    # that might exist in the source directory, e.g. when testing a sdist
+    # https://github.com/di/calver/issues/20
+    monkeypatch.setattr(
+        calver.integration, "get_pkginfo_contents", pretend.raiser(FileNotFoundError)
+    )
+
+
+@pytest.fixture
 def source_date_epoch(monkeypatch):
     _source_date_epoch = 1234567890
     monkeypatch.setenv("SOURCE_DATE_EPOCH", str(_source_date_epoch))
@@ -36,7 +46,7 @@ def test_version_missing(dist, keyword, original_version, value):
     assert dist.metadata.version == original_version
 
 
-def test_version_true(dist, keyword):
+def test_version_true(dist, keyword, ignore_pkginfo):
     value = True
 
     calver.integration.version(dist, keyword, value)
@@ -46,7 +56,7 @@ def test_version_true(dist, keyword):
     )
 
 
-def test_version_str(dist, keyword):
+def test_version_str(dist, keyword, ignore_pkginfo):
     value = "%c"
 
     calver.integration.version(dist, keyword, value)
@@ -54,7 +64,7 @@ def test_version_str(dist, keyword):
     assert dist.metadata.version == datetime.datetime.utcnow().strftime(value)
 
 
-def test_version_callable(dist, keyword):
+def test_version_callable(dist, keyword, ignore_pkginfo):
     v = pretend.stub()
     value = lambda: v
 
@@ -74,7 +84,7 @@ def test_reads_pkginfo(dist, keyword, monkeypatch):
     assert dist.metadata.version == "42"
 
 
-def test_reproducible_build(dist, keyword, source_date_epoch):
+def test_reproducible_build(dist, keyword, source_date_epoch, ignore_pkginfo):
     calver.integration.version(dist, keyword, True)
 
     assert dist.metadata.version == datetime.datetime.fromtimestamp(

--- a/tests/test_integration.py
+++ b/tests/test_integration.py
@@ -66,9 +66,8 @@ def test_version_str(dist, keyword, ignore_pkginfo):
 
 def test_version_callable(dist, keyword, ignore_pkginfo):
     v = pretend.stub()
-    value = lambda: v
 
-    calver.integration.version(dist, keyword, value)
+    calver.integration.version(dist, keyword, lambda: v)
 
     assert dist.metadata.version == v
 

